numba/cuda/tests/cudapy/test_optimization.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+
+from numba.cuda.testing import skip_on_cudasim, CUDATestCase
+from numba import cuda, float64
+import unittest
+
+
+def kernel_func(x):
+    x[0] = 1
+
+
+def device_func(x, y, z):
+    return x * y + z
+
+
+# Fragments of code that are removed from kernel_func's PTX when optimization
+# is on
+removed_by_opt = ( '__local_depot0', 'call.uni', 'st.param.b64')
+
+
+@skip_on_cudasim('Simulator does not optimize code')
+class TestOptimization(CUDATestCase):
+    def test_eager_opt(self):
+        # Optimization should occur by default
+        kernel = cuda.jit((float64[::1],))(kernel_func)
+        ptx = kernel.inspect_asm()
+
+        for fragment in removed_by_opt:
+            with self.subTest(fragment=fragment):
+                self.assertNotIn(fragment, ptx)
```

Comment:
Do these need to be subtests, it's quite heavy to just check for 3 strings in some disassembly and were it to fail the assert output would contain the failing string?

Reply:
I'm not sure what you mean by "heavy"... I thought it would be quicker to resolve a future fail that occurs due to changed code generation if the test output spelled out exactly which fragments were and were not found, instead of stopping at the first instance of a fragment not being found.

Reply:
I just mean that they are adding extra things into the testing framework where 3 is the size of the search space, balancing weight against statistical likelihood of occurrence and inconvenience in event of occurrence. I appreciate your point and am happy to leave given you prefer this :)